monkey/infection_monkey/exploit/log4shell_utils/ldap_server.py

```diff
@@ -202,20 +199,18 @@ def stop(self, timeout: Optional[float] = None):
                         terminates. If `timeout` is a positive floating point number, this method
                         blocks for at most `timeout` seconds.
         """
-        if self._server_process is None:
+        if self._server_thread is None:
             return
 
-        if self._server_process.is_alive():
+        if self._server_thread.is_alive():
             logger.debug("Stopping LDAP exploit server")
 
-            # The Twisted reactor registers signal handlers so it can catch SIGTERM and gracefully
-            # shutdown.
-            self._server_process.terminate()
-            self._server_process.join(timeout)
+            self._server_thread.join(timeout)
```

Comment:
There's a logical change here. The old code would send a signal to twisted (via SIGTERM) to shutdown. This code doesn't make any attempt to signal twisted to stop.